Bugzilla/Markdown.pm

```diff
@@ -0,0 +1,76 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Markdown;
+use 5.10.1;
+use Moo;
+
+use Encode;
+use Mojo::DOM;
+use HTML::Escape qw(escape_html);
+use List::MoreUtils qw(any);
+
+has 'markdown_parser' => (is => 'lazy');
+has 'bugzilla_shorthand' => (
+  is      => 'ro',
+  default => sub {
+    require Bugzilla::Template;
+    \&Bugzilla::Template::quoteUrls;
+  }
+);
+
+sub _build_markdown_parser {
+  if (Bugzilla->has_feature('alien_cmark')) {
+    require Bugzilla::Markdown::GFM;
+    require Bugzilla::Markdown::GFM::Parser;
+    return Bugzilla::Markdown::GFM::Parser->new(
+      {
+        hardbreaks => 1,
+        validate_utf8 => 1,
+        extensions => [qw( autolink tagfilter table strikethrough )],
+      }
+    );
+  }
+  else {
+    return undef;
+  }
+}
+
+sub render_html {
```

Comment:
Finally managed to figure out why quoteUrls wasn't being called at all, keen to get feedback here @dylanwh, thanks!